File: wiki-backend/src/main/java/com/internal/team/wiki/dto/doc/response/DocResponse.java
package com.internal.team.wiki.dto.doc.response;

import com.internal.team.wiki.domain.doc.DocEntity;
import lombok.Builder;
import lombok.Getter;

@Getter
public class DocResponse {

    private final String title;

    private final String contents;

    @Builder
    public DocResponse(final String title, final String contents) {
        this.title = title;
        this.contents = contents;
    }

    public static DocResponse from(final DocEntity docEntity) {
        return DocResponse.builder()
                .title(docEntity.getTitle().getValue())
                .contents(docEntity.getContents().getValue())
                .build();
    }
}

File: wiki-backend/src/main/java/com/internal/team/wiki/doc/dto/DocResponse.kt
package com.internal.team.wiki.doc.dto

import com.internal.team.wiki.doc.domain.DocEntity


data class DocResponse(
    val title: String,
    val contents: String
) {
    companion object {
        @JvmStatic
        fun from(docEntity: DocEntity): DocResponse {
            return DocResponse(
                title = docEntity.title.value,
                contents = docEntity.contents.value
            )
        }
    }
}

